spec/interaction/interaction_spec.js

```diff
@@ -1,134 +1,137 @@
-describe('Browser', function() {
+class Person {
 
-  var newBrowser;
+  constructor(name, browser) {
+    this.name = name;
+    this.browser = browser;
+    this.$ = browser.$;
+    this.element = browser.element;
+  }
 
-  afterEach(function(done) {
+  async openApp() {
+    await this.browser.get('index.html#/interaction');
+  };
+
+  async login() {
+    await this.element(by.model('userInput')).sendKeys(this.name);
+    await this.$('#sendUser').click();
+  };
+
+  async clearMessages() {
+    await this.$('#clearMessages').click();
+  };
+
+  async sendMessage(msg) {
+    await this.element(by.model('message')).sendKeys(msg);
+    await this.$('#sendMessage').click();
+  };
+
+  getMessages() {
+    return this.element.all(by.repeater('msg in messages track by $index'));
+  };
+};
+
+describe('Browser', () => {
+
+  let newBrowser;
+
+  afterEach(async() => {
     // Calling quit will remove the browser.
     // You can choose to not quit the browser, and protractor will quit all of
     // them for you when it exits (i.e. if you need a static number of browsers
     // throughout all of your tests). However, I'm forking browsers in my tests
     // and don't want to pile up my browser count.
     if (newBrowser) {
-      newBrowser.quit().then(() => {
-        done();
-      });
-    } else {
-      done();
+      await newBrowser.quit();
     }
   });
 
-  it('should be able to fork', function() {
-    browser.get('index.html');
-    newBrowser = browser.forkNewDriverInstance();
+  it('should be able to fork', async() => {
+    await browser.get('index.html');
+    newBrowser = await browser.forkNewDriverInstance().ready;
     expect(newBrowser).not.toEqual(browser);
     expect(newBrowser.driver).not.toEqual(browser.driver);
-    expect(newBrowser.driver.getCurrentUrl()).toEqual('data:,');
+    expect(await newBrowser.driver.getCurrentUrl()).toEqual('data:,');
   });
 
-  it('should be able to navigate to same url on fork', function() {
-    browser.get('index.html');
-    newBrowser = browser.forkNewDriverInstance(true);
-    expect(newBrowser.driver.getCurrentUrl()).
-        toMatch('index.html#/form');
+  it('should be able to navigate to same url on fork', async() => {
+    await browser.get('index.html');
+    newBrowser = await browser.forkNewDriverInstance(true).ready;
```

Comment:
Apparently, this comes from: https://github.com/angular/protractor/blob/ed955e56a839d7f69da43acb6755763220d3681d/lib/runner.ts#L275

I think this is so convoluted because it needed to support the control flow case as well. This should be cleaned up - forkNewDriverInstance() should just return a promise that resolves to the new browser instance, after its ready.